### DISCUSSION TITLE
Should throw exception when doesn't mapping correctly

### DISCUSSION THREAD

**Discussion by J2TeamNNL** [spatie/laravel-data#504] [Ideas]
When I create data using **from** function, it does not throw an exception even when incorrect information is passed, leading to unsuccessful mapping. Consequently, I can't ascertain the information is valid and filled.

**Comment by rubenvanassche** [comment]
Can you provide me an example?

**Reply by J2TeamNNL** [reply]
Here is an example
```php
class CreateTagData extends Data
{
    public function __construct(
        public string $name,
        public ?string $slug,
    ) {

    }
}

$arr = [
    'name' =>  'test',
    'name1' => 'test2',
];
$data = CreateTagData::from($arr);
```

dump $data we got this
```json
{
    "name": "test",
    "slug": null
}
```

I noticed that we have built-in function validateAndCreate but it's not cover this case

**Reply by J2TeamNNL** [reply]
I believe Laravel Data wants to replace the Request with rules, but it doesn't function like the all() method; instead, it uses validated(). 
I propose that we should have a method that throws an exception when something is passed without going through any validation.
As a developer, I expect that when I pass an array to Laravel Data, it will validate and transform all attributes of my array, not just retrieve the validated data.

**Reply by rubenvanassche** [reply]
> I believe Laravel Data wants to replace the Request with rules

And be a resource, be just a plain object, being a structured JSON object in a model, .... Data is a lot of things making it also a lot harder to generalize things.

> I propose that we should have a method that throws an exception when something is passed without going through any validation.

Not completely following here, `from` will never validate. There's been a lot of discussion about this, I know, but I'm still convinced data should not always validate. In most of the cases we're using data, we know data is validated so adding an extra validation step is useless and could even be harmful. We're planning to make this configurable in v4 though. 

As for your example, since your array contains a value which is not a property it does not validate, but its data is also not used. We had a feature for this in data-transfer-object. We could in the future implement such a thing in data. But at the moment I want to get everything what we have working 100% before adding even more complex new features.

**Reply by J2TeamNNL** [reply]
Thank you for the detailed explanation; I appreciate your insights. 
Regarding the discussion, I propose introducing a 'strict' attribute in the configuration while employing the 'from' method for data mapping. This approach would allow us to utilize the package not only for data transformation but also for validation purposes. By implementing such a feature, we can ensure that data integrity is maintained by enforcing validation rules where necessary, thereby enhancing the robustness of our data handling processes.

**Reply by rubenvanassche** [reply]
Take a look at `validation_strategy` in the config file, that's probably what you need.
